In [1]:
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from sklearn.model_selection import train_test_split

# Cargar datos
data = pd.read_csv('/content/drive/MyDrive/Tarea 7 Mineria de datos/datos.csv')

# Crear variable binaria objetivo
data['victima_trabaja_binaria'] = data['VIC_TRABAJA'].apply(lambda x: 1 if x == 1 else 0)

# Seleccionar variables predictoras
X = data[['VIC_ESCOLARIDAD', 'VIC_EST_CIV', 'AGR_ESCOLARIDAD', 'AGR_TRABAJA', 'AGR_EST_CIV']].fillna(0)
y = data['victima_trabaja_binaria']

# Dividir en train/test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Definir el modelo de red neuronal
model = Sequential()
model.add(Dense(8, input_dim=5, activation='relu'))
model.add(Dense(4, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

# Entrenamiento
history = model.fit(X_train, y_train, epochs=100, batch_size=64, validation_data=(X_test, y_test))

# Evaluación
loss, accuracy = model.evaluate(X_test, y_test)
print(f'Loss: {loss:.3f}, Accuracy: {accuracy:.3f}')


Epoch 1/100


/usr/local/lib/python3.12/dist-packages/keras/src/layers/core/dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


458/458 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.5963 - loss: 0.7461 - val_accuracy: 0.6285 - val_loss: 0.6609
Epoch 2/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6203 - loss: 0.6619 - val_accuracy: 0.6341 - val_loss: 0.6510
Epoch 3/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6253 - loss: 0.6522 - val_accuracy: 0.6459 - val_loss: 0.6515
Epoch 4/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6290 - loss: 0.6485 - val_accuracy: 0.6381 - val_loss: 0.6451
Epoch 5/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.6243 - loss: 0.6496 - val_accuracy: 0.6351 - val_loss: 0.6427
Epoch 6/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.6320 - loss: 0.6466 - val_accuracy: 0.6368 - val_loss: 0.6421
Epoch 7/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6291 - loss: 0.6466 - val_accuracy: 0.6422 - val_loss: 0.6411
Epoch 8/100
458/458 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6292 - loss: 0.6470 - val_accuracy: 0.6374

# Predicción

In [3]:
# Caso
nuevo_caso = np.array([[4, 2, 3, 1, 2]])  # [VIC_ESCOLARIDAD, VIC_EST_CIV, AGR_ESCOLARIDAD, AGR_TRABAJA, AGR_EST_CIV]
prediccion = model.predict(nuevo_caso)
print(prediccion)


1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 68ms/step
[[0.10912376]]


# Interpretación
La salida del modelo (0.1091) representa la probabilidad de que la víctima trabaje (1 = trabaja, 0 = no trabaja).

En este caso, el valor 0.109 (~10.9%) indica que el modelo predice que la víctima probablemente no trabaja.

# Conclusión


La red neuronal logró predecir correctamente la probabilidad de empleo de la víctima a partir de variables sociodemográficas y del agresor.
En este caso concreto:

Predicción: 0.109 → No trabaja
Confianza del modelo: ~89%
Precisión global del modelo: 68%
